RecoEgamma/EgammaTools/src/EffectiveAreas.cc

```diff
@@ -0,0 +1,119 @@
+#include "RecoEgamma/EgammaTools/interface/EffectiveAreas.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include <cmath>
+#include <fstream>
+#include <string>
+#include <sstream>
+
+EffectiveAreas::EffectiveAreas(const std::string& filename):
+  filename_(filename)
+{
+
+  // Open the file with the effective area constants
+  std::ifstream inputFile;
+  inputFile.open(filename_.c_str());
+  if( !inputFile.is_open() )
+    throw cms::Exception("EffectiveAreas config failure")
+      << "failed to open the file " << filename_ << std::endl;
+  
+  // Read file line by line
+  std::string line;
+  const float undef = -999;
+  while( getline(inputFile, line) ){
+    if(line[0]=='#') continue; // skip the comments lines
+    float etaMin = undef, etaMax = undef, effArea = undef;
+    std::stringstream ss(line);
+    ss >>  etaMin >> etaMax >> effArea;
+    // In case if the format is messed up, there are letters
+    // instead of numbers, or not exactly three numbers in the line,
+    // it is likely that one or more of these vars never changed
+    // the original "undef" value:
+    if( etaMin==undef || etaMax==undef || effArea==undef )
+      throw cms::Exception("EffectiveAreas config failure")
+	<< "wrong file format, file name " << filename_ << std::endl;
+    
+    absEtaMin_          .push_back( etaMin );
+    absEtaMax_          .push_back( etaMax );
+    effectiveAreaValues_.push_back( effArea );
+  }
+
+  // Extra consistency checks are in the function below.
+  // If any of them fail, an exception is thrown.
+  checkConsistency();
+}
+
+EffectiveAreas::~EffectiveAreas(){
+
+  absEtaMin_.clear();
+  absEtaMax_.clear();
+  effectiveAreaValues_.clear();
```

Comment:
These statements are not guaranteed to free the memory in the vectors. What needs to be done instead is:

vector<float>().swap(absEtaMin_);
and the same for the other two variables.